### PR TITLE
clean up code

### PR DESCRIPTION
**Pull request by nnym**
Switch from replacing Minecraft's Unbreaking registry entry by a custom enchantment to modifying only the required parts of the default enchantment through Mixin.
Update to Loom 0.9 and fix a minor build script issue.
When `config.maxLevelOnly`, check if the level of Unbreaking >= maximum level instead of == maximum level or == maximum level * 2.

The mod appears to behave as expected. Hopefully this pull request will also fix an incompatibility with my mod.

### CODE REVIEW

File: src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import wutdahack.actuallyunbreaking.AUConfig;
+
+@Mixin(UnbreakingEnchantment.class)
+abstract class UnbreakingEnchantmentMixin extends Enchantment {
+
+    // items can't have mending and unbreaking together
```
Comment (nnym) [inline src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:22]: Why is that? One might wish to enchant an unbreakable tool with lost durability with Mending in order to remove the damage bar.

Comment (wutdahack) [reply src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:22]: In the ItemStackMixin, I used the setDamage method in order to remove the damage bar, so there was no point to use Mending with Unbreaking.

Comment (nnym) [reply src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:22]: I see. I will fix that.

File: src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        } else {
+            ret.setReturnValue(1);
+        }
+    }
+
+    @Inject(method = "shouldPreventDamage", at = @At("HEAD"), cancellable = true)
+    private static void makeUnbreakable(ItemStack item, int level, Random random, CallbackInfoReturnable<Boolean> ret) {
+        if (ActuallyUnbreaking.isUnbreakable(level)) {
+            ret.setReturnValue(true);
```
Comment (wutdahack) [inline src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:43]: I think we could just use the setDamage method here using the item parameter.

Comment (nnym) [reply src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:43]: Here instead of in `ItemStack#damage`?

Comment (wutdahack) [reply src/main/java/wutdahack/actuallyunbreaking/mixin/UnbreakingEnchantmentMixin.java:43]: Yes.

**Review by wutdahack — APPROVED** [review]
Everything else seems to look good! Let's hope this fixes the incompatibility.

**Pull request merged**